### PR TITLE
fix(cni): pass experimental image version to install command

### LINKED ISSUE #4653: CNI: respect image version from command line
### What happened?

Running 

```bash
kumactl install control-plane --cni-enabled --cni-experimental --cni-version=0.0.0-preview.3a60ab8ba
```

will still result in trying to pull the current version (1.7.0). Since we are moving to helm `--set` values should this be fixed or let's just wait for everything to be ported to that way?

### PR DESCRIPTION
**Pull request by slonka**
Pass experimental image version to install command - fixes #4653 

Signed-off-by: slonka <slonka@users.noreply.github.com>

### Checklist prior to review

- [X] Link to docs PR or issue -- PR updating docs will be generated automatically
- [X] Link to UI issue or PR -- not a UI issue
- [X] Is the [issue worked on linked][1]? -- 
- [X] Unit Tests --
- [X] E2E Tests -- no e2e tests needed
- [X] Manual Universal Tests -- not a universal issue
- [x] Manual Kubernetes Tests -- done via `./kumactl install control-plane --set "experimental.cni=true" --set "cni.enabled=true" | grep image`
- [X] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? -- no
- [X] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? -- no

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kumactl/cmd/completion/testdata/bash.golden [hunk 1]
```diff
@@ -3430,8 +3430,6 @@ _kumactl_install_control-plane()
     local_nonpersistent_flags+=("--cni-conf-name=")
     flags+=("--cni-enabled")
     local_nonpersistent_flags+=("--cni-enabled")
-    flags+=("--cni-experimental")
```
Comment (slonka) [inline app/kumactl/cmd/completion/testdata/bash.golden:3433]: We agreed that we're depending on `--set` values so I'm removing this

File: app/kumactl/cmd/install/testdata/install-control-plane.cni-experimental-enabled.golden.yaml [hunk 2]
```diff
@@ -2035,7 +2035,7 @@ spec:
       terminationGracePeriodSeconds: 5
       containers:
         - name: install-cni
-          image: "docker.io/kumahq/kuma-cni:1.7.0"
+          image: "docker.io/kumahq/kuma-cni:0.0.1"
```
Comment (michaelbeaumont) [inline app/kumactl/cmd/install/testdata/install-control-plane.cni-experimental-enabled.golden.yaml:2038]: Where is `0.0.1` coming from? :thinking:

Comment (slonka) [reply app/kumactl/cmd/install/testdata/install-control-plane.cni-experimental-enabled.golden.yaml:2038]: https://github.com/kumahq/kuma/blob/f15239f7c13d0274b2387410f2bd8dd984327670/app/kumactl/cmd/install/install_control_plane_test.go#L31

Comment (michaelbeaumont) [reply app/kumactl/cmd/install/testdata/install-control-plane.cni-experimental-enabled.golden.yaml:2038]: ah nice, ok

**Review by michaelbeaumont — APPROVED** [review]

**Review by jakubdyszkiewicz — APPROVED** [review]

**Comment by michaelbeaumont** [comment]
@slonka Can this be merged?

**Comment by slonka** [comment]
I merged this, I'll resolve the conflict on the other PR

**Pull request merged**